client/components/foldable-card/index.js

```diff
@@ -0,0 +1,161 @@
+
+ import './style.scss';
+
+ class FoldableCard extends Component {
+     static displayName = 'FoldableCard';
+
+     static propTypes = {
+         actionButton: PropTypes.element,
+         actionButtonExpanded: PropTypes.element,
+         cardKey: PropTypes.string,
+         compact: PropTypes.bool,
+         disabled: PropTypes.bool,
+         expandedSummary: PropTypes.oneOfType( [ PropTypes.string, PropTypes.element ] ),
+         expanded: PropTypes.bool,
+         icon: PropTypes.string,
+         onClick: PropTypes.func,
+         onClose: PropTypes.func,
+         onOpen: PropTypes.func,
+         screenReaderText: PropTypes.oneOfType( [ PropTypes.string, PropTypes.bool ] ),
+         summary: PropTypes.oneOfType( [ PropTypes.string, PropTypes.element ] ),
+     };
+
+     static defaultProps = {
+         onOpen: noop,
+         onClose: noop,
+         cardKey: '',
+         icon: 'chevron-down',
+         expanded: false,
+         screenReaderText: false,
+     };
+
+     state = {
+         expanded: this.props.expanded,
+     };
+
+     componentWillReceiveProps( nextProps ) {
+         if ( nextProps.expanded !== this.props.expanded ) {
+             this.setState( { expanded: nextProps.expanded } );
+         }
+     }
+
+     onClick = () => {
+         if ( this.props.children ) {
+             this.setState( { expanded: ! this.state.expanded } );
+         }
+
+         if ( this.props.onClick ) {
+             this.props.onClick();
+         }
+
+         if ( this.state.expanded ) {
+             this.props.onClose( this.props.cardKey );
+         } else {
+             this.props.onOpen( this.props.cardKey );
+         }
+     };
+
+     getClickAction() {
+         if ( this.props.disabled ) {
+             return;
+         }
+         return this.onClick;
+     }
+
+     getActionButton() {
+         if ( this.state.expanded ) {
+             return this.props.actionButtonExpanded || this.props.actionButton;
+         }
+         return this.props.actionButton;
+     }
+
+     renderActionButton() {
+         const clickAction = ! this.props.clickableHeader ? this.getClickAction() : null;
+         if ( this.props.actionButton ) {
+             return (
+                 <div className="foldable-card__action" role="presentation" onClick={ clickAction }>
+                     { this.getActionButton() }
+                 </div>
+             );
+         }
+         if ( this.props.children ) {
+             const iconSize = 24;
+             const screenReaderText = this.props.screenReaderText || this.props.translate( 'More' );
+             return (
+                 <button
+                     disabled={ this.props.disabled }
+                     type="button"
+                     className="foldable-card__action foldable-card__expand"
+                     onClick={ clickAction }
+                 >
+                     <ScreenReaderText>{ screenReaderText }</ScreenReaderText>
```

Comment:
I debated rather or not to copy over Calypso's `<ScreenReaderText`> component. I ended up copying it because:
- it has less change compared to the existing submodule
- other Calypso components may also need it
- we will clean up this entire `client/components` folder so then it's not introducing long term tech debt.

Reply:
If there is a WordPress component for this, maybe we should switch over to it. If the goal is to remove Calypso components completely, then I think it is beneficial to have as little calypso component dependency as possible.

Reply:
Update: Removed ScreenReaderText component and use VisuallyHidden component instead.